Core/Src/main.c

```diff
@@ -95,6 +99,29 @@ int main(void)
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	  // send low to CS pin to start reading from ADC
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+
+	  // get data from ADC
+	  uint8_t transmit_data[3] = {0x01, 0x80, 0x00};
+	  uint8_t recieve_data[3];
+	  uint16_t size = 6;
+	  HAL_StatusTypeDef status = HAL_SPI_TransmitReceive(&hspi1, transmit_data, recieve_data, size, HAL_TIMEOUT);
+
+	  // send high to CS pin to stop reading from ADC
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+
+	  if (status == HAL_OK){
+		  // Get info from recieve_data and convert the 2 8-bit values into the final 16-bit value for count
+		  uint16_t value_from_adc = (recieve_data[1] << 8) | recieve_data[2];
+		  // Do AND comparison with 0x07FF to get the 10-bit count value
+		  uint16_t adc_count = 0x07FF;
+		  adc_count &= value_from_adc;
+
+		  // Set the compare register to the adc_count value computed previously
+		  __HAL_TIM_SET_COMPARE(&htim1, TIM_CHANNEL_1, adc_count);
```

Comment:
Ok so you have the adc counts, which is a value between 0 and 1023 bec it is a 10 bit ADC... but why would this be the timer compare value (hint, its not hehe)?

What the timer compare value does is set the value at which, when the timer counter reaches it, the output from the PWM signal will go from high to low, effectively setting your on time and effectively your duty cycle.

There are a couple things you have to consider when calculating this "compare val". First off, what "on time" is the application looking for? How many "timer ticks" corresponds to that "on time" and how does the adc value correlate to the percentage of the min to max on-time that needs to be set?

If you have any questions about this calculations please let me know!

Reply:
Sounds good, I think I have an idea now. 

Made it so the ADC value is converted to number of counts for the timer. Based on 1-2ms on-time and counter speed that I set in the timer to 3200 counts/ms

Equation is:
compare_val = (adc_count / 1023) * COUNTER_SPEED + COUNTER_SPEED;

Please let me know if this equation and logic is correct. Thanks!